### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by C-Otto**

### CODE REVIEW

**Comment by lrbalt** [comment]
There is a deprecation warning on Travis. Not sure if the other two errors are related to this pull, but the deprecation warning could be fixed.

**Comment by C-Otto** [comment]
The deprecation is handled in #1824.

**Comment by dnrce** [comment]
Needs a rebase

**Comment by C-Otto** [comment]
On what? I just merged master into this?

**Comment by dnrce** [comment]
Exactly -- rebasing onto master is cleaner than merging master in.

**Comment by dnrce** [comment]
I literally just mean the difference between `git merge upstream/master` and `git rebase upstream/master`.

**Comment by C-Otto** [comment]
OK, I guess this also holds for my other PRs?

**Comment by dnrce** [comment]
Where needed, it's always much cleaner for feature branches to be rebased than merged. But of course it usually isn't necessary to do either if there are no conflicts.

**Comment by C-Otto** [comment]
Sadly this drops all commit comments - and, of course, I have no idea when you merge which PR, and in which order...

**Comment by C-Otto** [comment]
Oh, and I merged because of the (now reverted) bugs.

**Comment by dnrce** [comment]
By "drops all commit comments" do you mean pull request comments on specific lines of the commits?

Comments get dropped if they were made directly on the commits in your branch. But that's the commenter's error. If the comments are made within the context of the PR, they are preserved, and GitHub adds a message like "so-and-so commented on an outdated diff" and provides the ability to show the earlier diff.

**Comment by C-Otto** [comment]
Yes. @lrbalt you're invited to comment in the PRs from now on :)

**Comment by C-Otto** [comment]
Could you have a look into this and merge, please?

File: features/step_definitions/project_steps.rb [hunk 1]
```diff
@@ -306,7 +306,7 @@
   expect(project).to_not be_nil
 end
 
-Then /^I should have (\d+) todo in project "([^"]*)"$/ do |todo_count, project_name|
+Then /^I should have (\d+) todo(?:s?) in project "([^"]*)"$/ do |todo_count, project_name|
```
Comment (dnrce) [inline features/step_definitions/project_steps.rb]: Can't this just be `todos?`?

Comment (C-Otto) [inline features/step_definitions/project_steps.rb]: No, it is also used for the singular version.

Comment (dnrce) [inline features/step_definitions/project_steps.rb]: Notice I wrote `todos?` with a question mark -- that should make the last character, the s, optional in the match.

Comment (C-Otto) [inline features/step_definitions/project_steps.rb]: Ah, I did not. Sure, that is simpler. Will do that later.

Comment (C-Otto) [inline features/step_definitions/project_steps.rb]: Fixed.

**Pull request merged**